api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreProperties.java

```diff
@@ -0,0 +1,44 @@
+package org.openmrs.module.registrationcore.api.impl;
+
+import org.openmrs.api.APIException;
+import org.openmrs.module.registrationcore.RegistrationCoreConstants;
+import org.openmrs.module.registrationcore.api.ModuleProperties;
+import org.openmrs.module.registrationcore.api.mpi.common.MpiProvider;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+public class RegistrationCoreProperties extends ModuleProperties implements ApplicationContextAware {
+    //TODO move getting properties related to Registration Core in this class.
+
+    private ApplicationContext applicationContext;
```

Comment:
@djaayeri I will try to use it.

On Wed, Aug 26, 2015 at 6:26 PM, Darius Jazayeri <notifications@github.com>
wrote:

> In
> api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreProperties.java
> <https://github.com/openmrs/openmrs-module-registrationcore/pull/11#discussion_r37995633>
> :
>
> > @@ -0,0 +1,44 @@
> > +package org.openmrs.module.registrationcore.api.impl;
> > +
> > +import org.openmrs.api.APIException;
> > +import org.openmrs.module.registrationcore.RegistrationCoreConstants;
> > +import org.openmrs.module.registrationcore.api.ModuleProperties;
> > +import org.openmrs.module.registrationcore.api.mpi.common.MpiProvider;
> > +import org.springframework.context.ApplicationContext;
> > +import org.springframework.context.ApplicationContextAware;
> > +
> > +public class RegistrationCoreProperties extends ModuleProperties implements ApplicationContextAware {
> > +    //TODO move getting properties related to Registration Core in this class.
> > +
> > +    private ApplicationContext applicationContext;
>
> By the way, OpenMRS has a Context.getRegisteredComponent() method which is
> equivalent to applicationContext.getBean(), so you wouldn't need to do
> ApplicationContextAware. This is fine though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openmrs/openmrs-module-registrationcore/pull/11/files#r37995633>
> .
>



-- 
-----------
Regards,
Roman Zayats